### PR TITLE
Feature/ex 18 mobile phone task exercise

### PR DESCRIPTION
**Pull request by KodzGS**

### CODE REVIEW

File: src/main/java/Wall.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+public class Wall {
```
Comment (owynowyn) [inline src/main/java/Wall.java:1]: Same as before mate, I think you've branched off your Wall exercise work for this branch, might wanna hop on a call for 5 minutes to sort this out at some point 👍

File: src/main/java/Contact.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+
+public class Contact {
```
Comment (owynowyn) [inline src/main/java/Contact.java:1]: Code is bang on bro, might just need an auto-format on this class, we can try and get something set up so you can do this a bit more intuitively too

Comment (owynowyn) [reply src/main/java/Contact.java:1]: When we can get on a call for 5

File: src/main/java/MobilePhone.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import java.util.ArrayList;
+import java.util.List;
+//test
```
Comment (owynowyn) [inline src/main/java/MobilePhone.java]: Sneaky comment - remove

File: src/main/java/MobilePhone.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        }
+        System.out.println("Contact doesn't exist");
+        return false;
+    }
+    /**
+     * findContact(), has one parameter of type Contact and returns an int.
+     * The returned value is it's position in the ArrayList, it will either be -1 (doesn't exists) or a value greater than or equal to 0 (does exists).
+     */
+    private int findContact(Contact contact) {
+        return this.myContacts.indexOf(contact);
```
Comment (owynowyn) [inline src/main/java/MobilePhone.java]: Small thing; probably dont need the `this` keyword here, doesn't change anything though, all looks good

File: src/main/java/MobilePhone.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    /**
+     * findContact(), same as above, only it has one parameter of type String.
+     */
+    public int findContact(String contact) {
+
+        if (myContacts.contains(myContacts.indexOf(contact))) {
```
Comment (owynowyn) [inline src/main/java/MobilePhone.java]: I dont think `myContacts.indexOf(contact)` will return what you expect here. The `myContacts.contain(Contact)` method, expects you to pass in a `Contact` object, but the `myContacts.indexOf(contact)` will return an int, so you will basically be calling `myContacts.contain(0)` for example. Which will probably result in an invalid argument exception as it expects `myContacts.contains(contact)`.
I would probably update this method to find the Contact within the list using the `String contact` parameter you have passed in, youll probably need a `for` loop for this. Give me a shout if you need some assistance with this or have a google

**Review by owynowyn — CHANGES_REQUESTED** [review]
Great work again mate, some adjustments to make to one of the methods but if we can get on a call this week to go through some of this then we should be in a really good place!

File: src/main/java/MobilePhone.java (hunk 3 above)
Comment (KodzGS) [reply src/main/java/MobilePhone.java]: removed

**Pull request merged**